spec/integration/indexer/indexing_spec.rb

```diff
@@ -125,4 +125,41 @@
     expect(last_response.status).to eq(202)
     expect_document_is_in_rummager(SAMPLE_DOCUMENT)
   end
+
+  it "rescheduling during locking does not corrupt the process" do
+    stubbed_client = client
+
+    locked_response = { "items" => [
+      { "index" => { "error" => { "reason" => "[FORBIDDEN/metasearch/index write" } } }
+    ] }
+
+    expect(stubbed_client).to receive(:bulk).and_return(locked_response)
```

Comment:
Did you consider any alternatives to this stubbing? Is it just difficult to write the test without stubbing because the Sidekiq retries are unpredictable?

I'm just worried that it makes the integration test more fragile. e.g. what happen if we upgrade Elasticsearch and the lock response changes?

Reply:
it might be possible but not sure how as in test mode sidekiq jobs run immediately. I guess we could lock the index run the test and have a separate thread that unlock the index after a small timeout?? but that is more complex